### ISSUE TITLE
Implement `h3o` - `h2o`'ing, Arrays of Hashes of Arrays ... recursively (bags 'o hash refs)

### ISSUE THREAD

**Issue by oodler577** [oodler577/p5-Util-H2O-More#5]
### Notes on implementation
* Interface - should accept all things h2o does [what about default accessors?]
* All hash refs should get accessors (what about default accessors?)
* all arrays to get an vmethod that returns all elements in it
* anything not ARRAY or HASH should be untouched
* add hint that _it's all HASH refs_, which will allow us to use `h2o -recurse` and be done with it (or make that the default and add a `-deep` flag to initiate an inspection of all paths in the data structure)

### SYNOPSIS

```
my $AoH = JSON::desode_json $http_client->content; # JSON that's an array of hashes or any mix of ARRAY and/or HASH
h3o $AoH;

foreach my $item (@{$AoH->_items}) {
  # $item has already had h2o applied to it
  ...
}

```

### Details
As frequent user of `Util::H2O::h2o` I'd like something like `-recurse` that does a full descent into the structure, even peeking into `ARRAY` refs, looking for `HASH` refs to _objectify_. Would imply `-recurse`; e.g., in the code below it'd be nice if `$HoAoH` ended up with the hash references contained in the `ARRAY` ref.
```#!/usr/bin/env perl

use strict;
use warnings;

use Util::H2O;
use Data::Dumper qw//;

my $HoA = {
  one => [qw/1 2 3 4 5/],
  two => [qw/6 7 8 9 0/],
};

h2o -recurse, $HoA;

print Data::Dumper::Dumper($HoA);

my $HoAoH = {
  one   => [qw/1 2 3 4 5/],
  two   => [qw/6 7 8 9 0/],
  three => [
     { four => 4, five => 5, six => 6 },
     { seven => 7, eight => 8, nine => 9 },
  ],
  ten => {
    eleven   => 11,
    twelve   => 12,
    thirteen => 13,
  },
};

h2o -recurse, $HoAoH;

print Data::Dumper::Dumper($HoAoH);
```

Just one level down, this would be super handy for `JSON` results from web APIs that return _Array_ based structure or that contain a mixture of record types. E.g., the `JSON` below requires treatment `foreach` after the `JSON::decode_json`.

```
[
  {
    "userId": 1,
    "id": 1,
    "title": "delectus aut autem",
    "completed": false
  },
  {
    "userId": 1,
    "id": 2,
    "title": "quis ut nam facilis et officia qui",
    "completed": false
  },
  {
    "userId": 1,
    "id": 3,
    "title": "fugiat veniam minus",
    "completed": false
  },
  {
    "userId": 1,
    "id": 4,
    "title": "et porro tempora",
    "completed": true
  },
  {
    "userId": 1,
    "id": 5,
    "title": "laboriosam mollitia et enim quasi adipisci quia provident illum",
    "completed": false
  },
  {
    "userId": 1,
    "id": 6,
    "title": "qui ullam ratione quibusdam voluptatem quia omnis",
    "completed": false
  },
  {
    "userId": 1,
    "id": 7,
    "title": "illo expedita consequatur quia in",
    "completed": false
  },
  {
    "userId": 1,
    "id": 8,
    "title": "quo adipisci enim quam ut ab",
    "completed": true
  },
  {
    "userId": 1,
    "id": 9,
    "title": "molestiae perspiciatis ipsa",
    "completed": false
  },
  {
    "userId": 1,
    "id": 10,
    "title": "illo est ratione doloremque quia maiores aut",
    "completed": true
  },
  {
    "userId": 1,
    "id": 11,
    "title": "vero rerum temporibus dolor",
    "completed": true
  },
  {
    "userId": 1,
    "id": 12,
    "title": "ipsa repellendus fugit nisi",
    "completed": true
  },
  {
    "userId": 1,
    "id": 13,
    "title": "et doloremque nulla",
    "completed": false
  },
  {
    "userId": 1,
    "id": 14,
    "title": "repellendus sunt dolores architecto voluptatum",
    "completed": true
  },
  {
    "userId": 1,
    "id": 15,
    "title": "ab voluptatum amet voluptas",
    "completed": true
  },
  {
    "userId": 1,
    "id": 16,
    "title": "accusamus eos facilis sint et aut voluptatem",
    "completed": true
  },
  {
    "userId": 1,
    "id": 17,
    "title": "quo laboriosam deleniti aut qui",
    "completed": true
  },
  {
    "userId": 1,
    "id": 18,
    "title": "dolorum est consequatur ea mollitia in culpa",
    "completed": false
  }
]
```

**Comment by oodler577** [comment]
Plus support to _undo_ like `o2h` provides.

**Comment by oodler577** [comment]
call it `h3o` and `o3h` ( _heavy_ water)

**Comment by perlsheperd** [comment]
released via 0.1

**Comment by oodler577** [comment]
Done.
